ocaml/tests/test_clustering.ml

```diff
@@ -364,9 +364,40 @@ let test_disallow_unplug_with_clustering () =
   check_disallow_unplug true __context pif
     "PIF.set_disallow_unplug should be idempotent even with clustering"
 
-let test_disallow_unplug_ro_with_clustering_enabled =
+let test_assert_no_clustering_on_pif () =
+  let __context = T.make_test_database () in
+  let host, _, self = make_host_network_pif ~__context in
+  let assert_no_clustering_on self msg =
+    Alcotest.(check unit) msg
+      () (Xapi_pif.assert_no_clustering_enabled_on ~__context ~self)
+  in
+  assert_no_clustering_on self
+    "assert_no_clustering_on_pif without clustering";
+
+  (* Add an enabled cluster_host *)
+  let cluster, cluster_host = T.make_cluster_and_cluster_host ~__context ~host ~pIF:self () in
+  Alcotest.check_raises
+    "Live cluster_host on PIF"
+    Api_errors.(Server_error (clustering_enabled, [ Ref.string_of cluster_host ]))
+    (fun () -> Xapi_pif.assert_no_clustering_enabled_on ~__context ~self);
+
+  (* Disable clustering on PIF *)
+  Db.Cluster_host.set_enabled ~__context ~self:cluster_host ~value:false;
+  assert_no_clustering_on self
+    "assert_no_clustering_on_pif with clustering disabled";
+
+  for i = 0 to 5 do
+    T.make_cluster_host ~__context ~cluster ~pIF:self () |> ignore
+  done;
+  Alcotest.check_raises
+    "should fail with multiple cluster hosts on PIF"
+    Api_errors.(Server_error (cluster_does_not_have_one_node, [ "7" ]))
```

Comment:
This is unrelated to your PR, but I found this error a but confusing, because if multiple Cluster_hosts share the same PIF, then they must belong to different clusters. But this is an issue in the assertion itself.